src/main/java/com/ifactory/press/db/solr/spelling/suggest/MultiSuggester.java

```diff
@@ -252,60 +265,92 @@ private void initWeights(NamedList fieldConfigs, SolrCore coreParam) {
   public void build(SolrCore coreParam, SolrIndexSearcher searcher) throws IOException {
     LOG.info("build suggestion index: " + name);
     reader = searcher.getIndexReader();
+    long startTime = System.currentTimeMillis();
 
     SafariInfixSuggester ais = (SafariInfixSuggester) lookup;
     ais.clear();
-    
+
+    // index all stored fields with one pass through the index
+    buildFromStoredFields(storedFields, searcher);
+
     // index all the terms-based fields using dictionaries
-    for (WeightedField fld : fields) {
-      if (fld.useStoredField) {
-        buildFromStoredField(fld, searcher);
-      } else {
-        // TODO: refactor b/c we're not really using the MultiDictionary's multiple dictionary capability any more
-        dictionary = new MultiDictionary();
-        buildFromTerms(fld);
-        ais.add(dictionary);
-        ais.refresh();
-      }
+    for (WeightedField termField : termFields) {
+      // TODO: refactor b/c we're not really using the MultiDictionary's multiple dictionary capability any more
+      dictionary = new MultiDictionary();
+      buildFromTerms(termField);
+      ais.add(dictionary);
+      ais.refresh();
     }
-    LOG.info(String.format("%s suggestion index built: %d suggestions", name, ais.getCount()));
+
+    long endTime = System.currentTimeMillis();
+    long totalBuildTime = (endTime - startTime) / 1000;
+    LOG.info(String.format("%s suggestion index built: %d suggestions. Total time: %d seconds.", name, ais.getCount(), totalBuildTime));
   }
 
-  private void buildFromStoredField(WeightedField fld, SolrIndexSearcher searcher) throws IOException {
-    if (fld.fieldAnalyzer != null) {
-      throw new IllegalStateException("not supported: analyzing stored fields");
-    }
-    LOG.info(String.format("build suggestions from values for: %s (%d)", fld.fieldName, fld.weight));
-    Set<String> fieldsToLoad = new HashSet<String>();
-    fieldsToLoad.add(fld.fieldName);
+  private void buildFromStoredFields(List<WeightedField> flds, SolrIndexSearcher searcher) throws IOException {
+    LOG.info(String.format("%s Suggest BUILD for stored fields: %s", name, flds));
+    long startTime = System.currentTimeMillis();
+    long endTime;
+    long elapsedSeconds;
+    int lastCommitCount = 0;
+    addedStoredValues = new HashSet<>();
+    Set<String> fieldsToLoad = flds.stream().map(fld -> fld.fieldName).collect(Collectors.toSet());
     // Load the format field so we can filter out docs by specific formats
     if(this.shouldExcludeFormats) {
       fieldsToLoad.add("format");
     }
     int maxDoc = searcher.maxDoc();
```

Comment:
I was contemplating the other day, if we could just do searches on `searcher`, rather than grabbing `maxDoc`, to only grab the documents we're interested in. For instance, for ISBNs, we'd only need to grab documents with `content_type=book`, e.g. books and not chapters.

Something to consider for later, as I don't think it's necessary now at all.

Reply:
If this were possible, it would seem like we could also just use `fq=-format:collection` or similar to retrieve non collections/playlists so that we don't have to manually skip them below.